react-front-end/tsrc/theme/ThemePage.tsx

```diff
@@ -190,68 +191,37 @@ export const ThemePage = ({
     }
   };
 
-  const submitTheme = (themeSettings: IThemeSettings) =>
-    axios.put<IThemeSettings>(`${API_BASE_URL}/theme/settings/`, themeSettings);
+  const submitTheme = async (themeSettings: OEQ.Theme.ThemeSettings) =>
+    await OEQ.Theme.updateThemeSettings(API_BASE_URL, themeSettings);
 
-  const submitLogo = () =>
+  const submitLogo = async () =>
     logoSettings.logoToUpload &&
-    axios.put(`${API_BASE_URL}/theme/logo/`, logoSettings.logoToUpload);
+    (await OEQ.Theme.updateThemeLogo(API_BASE_URL, logoSettings.logoToUpload));
 
   const saveChanges = async () => {
     try {
-      await submitLogo();
       await submitTheme(mapColorsToSettings(themeColors));
+      await submitLogo();
+      setIsChangesUnsaved(false);
       setIsChangesUnsaved(false);
-      setIsShowSuccess(true);
       reload();
     } catch (error) {
-      if (isAxiosError(error)) {
-        handleError(error);
-      } else {
-        console.error("Unexpected non-Error caught in saveChanges(): " + error);
-      }
+      setError(
+        error instanceof Error
+          ? error
+          : new Error(`Unexpected non-Error caught in saveChanges(): $(error)`)
```

Comment:
Whoops, you've got a typo there:

```suggestion
          : new Error(`Unexpected non-Error caught in saveChanges(): ${error}`)
```